app/controllers/optIn/ChooseYearController.scala

```diff
@@ -0,0 +1,95 @@
+import play.api.i18n.I18nSupport
+import play.api.mvc._
+import services.optIn.OptInService
+import utils.AuthenticatorPredicate
+import views.html.optIn.ChooseTaxYearView
+
+import javax.inject.Inject
+import scala.concurrent.{ExecutionContext, Future}
+
+class ChooseYearController @Inject()(val optInService: OptInService,
+                                     val view: ChooseTaxYearView,
+                                     val authorisedFunctions: FrontendAuthorisedFunctions,
+                                     val auth: AuthenticatorPredicate)
+                                    (implicit val appConfig: FrontendAppConfig,
+                                          mcc: MessagesControllerComponents,
+                                          val ec: ExecutionContext,
+                                          val itvcErrorHandler: ItvcErrorHandler,
+                                          val itvcErrorHandlerAgent: AgentItvcErrorHandler)
+  extends ClientConfirmedController with FeatureSwitching with I18nSupport {
+
+  private val errorHandler = (isAgent: Boolean) => if (isAgent) itvcErrorHandlerAgent else itvcErrorHandler
+
+  private def withRecover(isAgent: Boolean)(code: => Future[Result])(implicit mtdItUser: MtdItUser[_]): Future[Result] = {
+    code.recover {
+      case ex: Exception =>
+        Logger("application").error(s"request failed :: $ex")
+        errorHandler(isAgent).showInternalServerError()
+    }
+  }
+
+  def show(isAgent: Boolean = false): Action[AnyContent] = auth.authenticatedAction(isAgent) {
+    implicit user =>
+      withRecover(isAgent) {
+        Future.successful(Ok(view(ChooseTaxYearForm(optInService.availableOptInTaxYear().map(_.toString)), viewModel(isAgent))))
+      }
+  }
+
+  def submit(isAgent: Boolean): Action[AnyContent] = auth.authenticatedAction(isAgent) {
+    implicit user =>
+      ChooseTaxYearForm(optInService.availableOptInTaxYear().map(_.toString)).bindFromRequest().fold(
+        formWithError => Future.successful(BadRequest(view(formWithError, viewModel(isAgent)))),
+        form => saveTaxYearChoice(form).map {
+          case true => redirectToCheckpointPage(isAgent)
+          case false => itvcErrorHandler.showInternalServerError()
+        }
+      )
+  }
+
+  private def saveTaxYearChoice(form: ChooseTaxYearForm)(implicit request: RequestHeader): Future[Boolean] = {
+    form.choice.flatMap(strFormat => TaxYear.getTaxYearModel(strFormat)).map { intent =>
+      optInService.saveIntent(intent)
+    } getOrElse Future.failed(new RuntimeException("no tax-year choice available"))
+  }
+
+  private def redirectToCheckpointPage(isAgent: Boolean): Result = {
+    val nextPage = controllers.optIn.routes.CheckYourAnswersController.show(isAgent)
+    Logger("application").info(s"redirecting to : $nextPage")
+    Redirect(nextPage)
+  }
+
+  private def cancelUrl(isAgent: Boolean): String = {
+    routes.ChooseYearController.show(isAgent).url //todo change this to the correct url
```

Comment:
Is it possible to put a placeholder in for now?

Reply:
this is the placeholder, but what do you suggest?

Reply:
I meant a placeholder for the page it should go to, a bit like you've done with the Check Your Answers page. That way it won't get accidentally left behind when that page become available.

Reply:
Okay, so what is the page it should go to?

Reply:
Ah... good question. I thought the ticket or prototype would say. I suspect it's the new Your Reporting Frequency page, but we can check at stand-up.

Reply:
done